core/src/main/java/com/datastax/oss/driver/api/core/uuid/Uuids.java

```diff
@@ -202,13 +235,188 @@ private static long makeClockSeqAndNode() {
+   * namespace UUID and the name parts concatenated together.
+   *
+   * <p>The byte array is expected to be at least 16 bytes long.
+   *
+   * @param namespaceAndName A byte array containing the concatenated namespace UUID and name;
+   *     cannot be null.
+   * @throws NullPointerException if <code>namespaceAndName</code> is null.
+   * @throws IllegalArgumentException if <code>namespaceAndName</code> is not at least 16 bytes
+   *     long.
+   * @throws IllegalStateException if the {@link MessageDigest} algorithm for version 3 (MD5) is not
+   *     available on this platform.
+   */
+  @NonNull
+  public static UUID nameBased(@NonNull byte[] namespaceAndName) {
+    return nameBased(namespaceAndName, 3);
+  }
+
+  /**
+   * Creates a new name-based (version 3 or version 5) {@link UUID} from the given byte array
+   * containing the namespace UUID and the name parts concatenated together.
+   *
+   * <p>The byte array is expected to be at least 16 bytes long.
+   *
+   * @param namespaceAndName A byte array containing the concatenated namespace UUID and name;
+   *     cannot be null.
+   * @param version The version to use, must be either 3 or 5.
+   * @throws NullPointerException if <code>namespaceAndName</code> is null.
+   * @throws IllegalArgumentException if <code>namespaceAndName</code> is not at least 16 bytes
+   *     long.
+   * @throws IllegalArgumentException if <code>version</code> is not 3 nor 5.
+   * @throws IllegalStateException if the {@link MessageDigest} algorithm for the desired version is
+   *     not available on this platform.
+   */
+  @NonNull
+  public static UUID nameBased(@NonNull byte[] namespaceAndName, int version) {
+    Objects.requireNonNull(namespaceAndName, "namespaceAndName cannot be null");
+    if (namespaceAndName.length < 16) {
+      throw new IllegalArgumentException("namespaceAndName must be at least 16 bytes long");
+    }
+    MessageDigest md = newMessageDigest(version);
+    md.update(namespaceAndName);
+    return buildNamedUuid(md.digest(), version);
+  }
+
+  @NonNull
+  private static MessageDigest newMessageDigest(int version) {
+    if (version != 3 && version != 5) {
+      throw new IllegalArgumentException(
+          "Invalid name-based UUID version, expecting 3 or 5, got: " + version);
+    }
+    String algorithm = version == 3 ? "MD5" : "SHA-1";
+    try {
+      return MessageDigest.getInstance(algorithm);
+    } catch (NoSuchAlgorithmException e) {
+      throw new IllegalStateException(algorithm + " algorithm not available", e);
+    }
+  }
+
+  @NonNull
+  private static UUID buildNamedUuid(@NonNull byte[] data, int version) {
+    // clear and set version
+    data[6] &= (byte) 0x0f;
+    data[6] |= (byte) (version << 4);
+    // clear and set variant to IETF
+    data[8] &= (byte) 0x3f;
+    data[8] |= (byte) 0x80;
+    return fromBytes(data);
+  }
+
+  private static UUID fromBytes(byte[] data) {
```

Comment:
I wonder if it would be useful to expose this method and the one below as generic utilities to convert to/from UUID and byte arrays?

Reply:
IMO this is too low-level and specific to be provided as a driver feature.